tests/Unit/Http/Controllers/RateControllerTest.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Tests\Unit\Http\Controllers;
+
+use App\Http\Controllers\Frontend\RateController;
+use App\Models\Rate;
+use App\Models\Movie;
+use App\Models\User;
+use Tests\TestCase;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Database\Eloquent\Collection;
+use App\Repositories\RateRepositoryInterface;
+use Mockery;
+use Illuminate\Http\Request;
+use Illuminate\Foundation\Testing\WithoutMiddleware;
+
+class RateControllerTest extends TestCase
```

Comment:
em có những testcase gì trong toàn bộ file test này?

Reply:
test_store_rate_with_oldrate: e test việc thêm mới rate khi không có giá trị rate cũ nào
test_store_rate_with_newrate: e test việc thêm mới rate khi đã tồn tại giá trị rate cũ

Reply:
setUp dùng để chạy trước khi bắt đầu function test
tearDown dùng để thiết lập lại mọi thứ trở về trạng thái trước khi function chạy, chạy sau khi kết thúc 1 function